src/components/article/AllArticles.jsx

```diff
@@ -26,12 +29,31 @@ class AllArticles extends Component {
     this.setState({ isFetching: false });
   };
 
+  handleBookmarkClick = ({ id, slug, bookmarked }) => {
+    const {
+      bookmarkArticle: bookMark,
+      unbookmarkArticle: unbookMark,
+      isLoggedIn,
+    } = this.props;
+    const articlePage = 'allArticles';
+    if (isLoggedIn === false) {
+      return toastr.error('Please login or signup');
+    }
+    return bookmarked ? unbookMark(id, articlePage) : bookMark(slug, articlePage, id);
+  }
+
   renderArticles() {
     const { allArticles } = this.props;
     const { isFetching } = this.state;
 
     if (!isFetching) {
-      return allArticles.data.map(article => <ArticleCard key={article.id} article={article} />);
+      return (allArticles.data.map(article => (
+        <ArticleCard
+          key={article.id}
+          handleBookmarkClick={event => this.handleBookmarkClick(article, event)}
```

Comment:
There's actually no need to pass in the `event` argument here since you are making no use of it.

Reply:
nice observation @akhilome taken care of that now